[nativert] make runtime const folding aware of run_const_graph

dolpm opened the pull request:
Summary: it's possible that we have foldable nodes that use things that will be folded by run_const_graph

Test Plan:
CI

Rollback Plan:

Differential Revision: D80355542

SherlockNoMad reviewed (commented):
test cases?

SherlockNoMad commented:
tbh, I would rather have these two run_const_graph have NO interaction. 
we can make them mutually exclusive.

dolpm commented:
> tbh, I would rather have these two run_const_graph have NO interaction. we can make them mutually exclusive.

i agree with this, but there are some cases where it could help. if we are doing graph mutations on things that use folded values (from run_const_graph), then it is possible that this opens up more opportunities.

dolpm commented:
> test cases?

i spot checked it on a model i was working with -- i can add a case.

dolpm committed:
[nativert] make runtime const folding aware of run_const_graph (#160760)

Summary:

it's possible that we have foldable nodes that use things that will be folded by run_const_graph

Test Plan:
CI
---
```
buck2 test //sigmoid/inference/test:test_passes -- test_const_folding_pass_with_runtime_folding --print-passing-details
```
```
I0820 11:13:58.399837 1205583 ConstantFolder.cpp:61] run_const_graph_node: true

I0820 11:13:58.399842 1205583 ConstantFolder.cpp:130] Const-folded node: %add_scalar = torch.ops.aten.add.Scalar(self=%getitem, other=0)
I0820 11:13:58.399844 1205583 ConstantFolder.cpp:130] Const-folded node: %v0[] = prim.ListPack(l0=%c_w_pre, l1=%c_b)
```
graph before:
```
graph(%c_w_pre, %c_b):
%getitem = torch.ops.higher_order.run_const_graph(=%v0[], =<subgraph>)
%add_scalar = torch.ops.aten.add.Scalar(self=%getitem, other=0)
return(%add_scalar)
```
graph after:
```
graph(%c_w_pre, %c_b):
%getitem = torch.ops.higher_order.run_const_graph(=%v0[], =<subgraph>)
return(%add_scalar)
```

Rollback Plan:

Reviewed By: SherlockNoMad

Differential Revision: D80355542
